Digital Credentials: remove outdated OpenID4VP infrastructure

marcoscaceres opened the pull request:
#### 0ff524063083836f63189b8eaec592a3a460b1e9
<pre>
Digital Credentials: remove outdated OpenID4VP infrastructure
<a href="https://rdar.apple.com/150654607">rdar://150654607</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=292129">https://bugs.webkit.org/show_bug.cgi?id=292129</a>

Reviewed by Anne van Kesteren.

Removed OpenID4VP-related code, as we don&apos;t make use of the stubs and the protocol identifiers has changed.

Test: http/wpt/identity/formats/openid/openid4vp.https.html

* LayoutTests/http/wpt/identity/formats/ISO18013/mobiledocument.https.html:
* LayoutTests/http/wpt/identity/formats/openid/openid4vp.https-expected.txt: Added.
* LayoutTests/http/wpt/identity/formats/openid/openid4vp.https.html: Added.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https.html:
* LayoutTests/platform/mac-sonoma/TestExpectations:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/CredentialRequestCoordinatorClient.h:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
(WebCore::jsToCredentialRequest):
* Source/WebCore/Modules/identity/DigitalCredentialsRequestData.h:
* Source/WebCore/Modules/identity/IdentityCredentialProtocol.h:
* Source/WebCore/Modules/identity/IdentityCredentialProtocol.idl:
* Source/WebCore/Modules/identity/protocols/DigitalCredentialsProtocols.h:
* Source/WebCore/Modules/identity/protocols/UnvalidatedDigitalCredentialRequest.h:
* Source/WebCore/Modules/identity/protocols/openid/OpenID4VPRequest.h: Removed.
* Source/WebCore/Modules/identity/protocols/openid/OpenID4VPRequest.idl: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/Chrome.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:
(-[WKDigitalCredentialsPicker fetchRawRequestsWithCompletionHandler:]):
(-[WKDigitalCredentialsPicker performRequest:]):
(-[WKDigitalCredentialsPicker handlePresentmentCompletionWithResponse:error:]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::fetchRawDigitalCredentialRequests):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.h:
* Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/cocoa/IdentityDocumentServices/DigitalCredentialsRequestValidatorBridge.h:
* Source/WebKit/WebProcess/cocoa/IdentityDocumentServices/DigitalCredentialsRequestValidatorBridge.mm:
(WebKit::DigitalCredentials::validateRequests):

Canonical link: <a href="https://commits.webkit.org/305311@main">https://commits.webkit.org/305311@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14ec6e988b518a161fd3743e8fa3e427b66038e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/10358 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/49374 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/146062 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90968 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/47a84cb0-7b62-4390-8b57-6886ce66221d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/139866 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/11059 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/10504 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/105531 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/77011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/8484bc36-817b-4c72-b869-fa331057de2b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140938 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/8251 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/123714 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/86381 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/aaedb644-5b52-4b5e-b958-b854ecd917d7) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7865 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/5620 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/6344 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/117262 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41890 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/148772 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/10040 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/42449 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113933 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/10057 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/8477 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/114264 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/29052 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7804 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119977 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/64764 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/10086 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37958 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9817 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/73654 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/10027 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9878 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:
```diff
@@ -304,12 +299,7 @@ - (void)performRequest:(const WebCore::DigitalCredentialsRequestData &)requestDa
-        }
-
-        auto validatedRequest = std::get<WebCore::ValidatedMobileDocumentRequest>(request);
+        auto validatedRequest = request;
```
This seems unneeded. Either rename the variable above or below.

annevk commented on Source/WebKit/WebProcess/cocoa/IdentityDocumentServices/DigitalCredentialsRequestValidatorBridge.mm:
```diff
@@ -147,28 +147,15 @@
-        }
-
-        auto mobileDocumentRequest = std::get<WebCore::MobileDocumentRequest>(request);
+        auto mobileDocumentRequest = request;
```
Ditto.

annevk reviewed (approved):
Looks good modulo nits.

marcoscaceres committed:
Digital Credentials: remove outdated OpenID4VP infrastructure
rdar://150654607
https://bugs.webkit.org/show_bug.cgi?id=292129

Reviewed by Anne van Kesteren.

Removed OpenID4VP-related code, as we don't make use of the stubs and the protocol identifiers has changed.

Test: http/wpt/identity/formats/openid/openid4vp.https.html

* LayoutTests/http/wpt/identity/formats/ISO18013/mobiledocument.https.html:
* LayoutTests/http/wpt/identity/formats/openid/openid4vp.https-expected.txt: Added.
* LayoutTests/http/wpt/identity/formats/openid/openid4vp.https.html: Added.
* LayoutTests/http/wpt/identity/identitycredentialscontainer-get-hidden.https.html:
* LayoutTests/platform/mac-sonoma/TestExpectations:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/CredentialRequestCoordinatorClient.h:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
(WebCore::jsToCredentialRequest):
* Source/WebCore/Modules/identity/DigitalCredentialsRequestData.h:
* Source/WebCore/Modules/identity/IdentityCredentialProtocol.h:
* Source/WebCore/Modules/identity/IdentityCredentialProtocol.idl:
* Source/WebCore/Modules/identity/protocols/DigitalCredentialsProtocols.h:
* Source/WebCore/Modules/identity/protocols/UnvalidatedDigitalCredentialRequest.h:
* Source/WebCore/Modules/identity/protocols/openid/OpenID4VPRequest.h: Removed.
* Source/WebCore/Modules/identity/protocols/openid/OpenID4VPRequest.idl: Removed.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/Chrome.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.h:
* Source/WebKit/UIProcess/DigitalCredentials/WKDigitalCredentialsPicker.mm:
(-[WKDigitalCredentialsPicker fetchRawRequestsWithCompletionHandler:]):
(-[WKDigitalCredentialsPicker performRequest:]):
(-[WKDigitalCredentialsPicker handlePresentmentCompletionWithResponse:error:]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::fetchRawDigitalCredentialRequests):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.h:
* Source/WebKit/WebProcess/DigitalCredentials/DigitalCredentialsCoordinator.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/cocoa/IdentityDocumentServices/DigitalCredentialsRequestValidatorBridge.h:
* Source/WebKit/WebProcess/cocoa/IdentityDocumentServices/DigitalCredentialsRequestValidatorBridge.mm:
(WebKit::DigitalCredentials::validateRequests):

Canonical link: https://commits.webkit.org/305311@main
